client/js/widgets/scoreboard.js

```diff
@@ -0,0 +1,287 @@
+import { $, $a, removeFromDOM, asArray, escapeID } from '../domhelpers.js';
+import { Widget } from './widget.js';
+
+class ScoreBoard extends Widget {
+  constructor(object, surface) {
+    super(object, surface);
+
+    this.addDefaults({
+      movable: true,
+      width: 300,
+      height: 200,
+      layer: -1,
+      typeClasses: 'widget scoreboard',
+      usePlayerColors: true,
+      playersInColumns: true,
+      rounds: null,
+      sortField: 'index',
+      sortAscending: true,
+      scoreProperty: 'score',
+      includeAllSeats: false,
+      showTotals: true,
+      seats: null,
+      showAllRounds: false,
+      roundLabel: 'Round',
+      currentRound: null
+    });
+  }
+
+  applyDeltaToDOM(delta) {
+    if(delta) // Don't call super unless this is a real delta to the scoreboard rather than from a seat
+      super.applyDeltaToDOM(delta);
+    const seats = this.getIncludedSeats();
+    if(seats !== null)
```

Comment:
This doesn't feel right. If `seats === null` after it wasn't previously, this would display the old table for any client that was connected with the old state but I assume that newly connected clients wouldn't see anything on the widget.

It should at least be consistent behavior. Maybe just show an error message on the widget? Or if it's not considered an error, just clear the widget for everyone?

Reply:
`seats` can be null after this call (for example, a property was changed in a seat where nobody is sitting, and `includeAllSeats == false` in this scoreboard. But I agree about clearing the widget.